### PR TITLE
Leandro/precommit blog script

### PR DESCRIPTION
**Pull request by leandrogavidia**
## Summary by Sourcery

Add a script to generate CMS dictionaries with short links for blog articles and integrate Husky for pre-commit hooks to enhance code quality checks. Modify the generateShortLink function to optionally bypass cookie usage.

New Features:
- Introduce a script to generate CMS dictionaries with short links for blog articles.

Enhancements:
- Modify the generateShortLink function to optionally bypass cookie usage.

Build:
- Add Husky as a dependency to manage Git hooks.

CI:
- Set up a pre-commit hook using Husky to ensure code quality before commits.

## Summary by CodeRabbit

- **New Features**
	- Added multiple new blog entries discussing various aspects of large language models (LLMs), including their applications, challenges, and advancements in AI.
  
- **Improvements**
	- Enhanced readability of JSON structures by formatting "topics" arrays into multi-line formats.
	- Introduced a "shortLink" field in various blog entries for potential future use.

- **Documentation**
	- Updated blog content with structured sections for better organization and user engagement.

### CODE REVIEW

File: apps/web/scripts/generate-cms-dictionaries.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import path from "node:path";
+import * as fs from "fs";
+
+async function main() {
+  AVAILABLE_LANGS.map(async (lang) => {
+    let categories = (await getArticleSections(lang as Lang)).map(
+      (section) => section.slug
+    );
+
+    categories.map(async (category) => {
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/web/scripts/generate-cms-dictionaries.ts:19]: @leandrogavidia please apply suggested improvement. Maybe not literally as the suggestion but to use a `Promise.all` for all of this chaining requests.

File: apps/web/scripts/generate-cms-dictionaries.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        await getBlogArticleData(lang as Lang, category, slug);
+        await addShortLink({
+          category,
+          lang: lang as Lang,
+          slug,
+        });
+      });
+    });
+  });
+}
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/web/scripts/generate-cms-dictionaries.ts:35]: @leandrogavidia same as previous comment, to use `Promise.all` to catch all chaining request properly :eyes:

File: apps/web/scripts/generate-cms-dictionaries.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    json["shortLink"] = shortLink || "";
+    const updatedJson = JSON.stringify(json, null, 2);
+    await fs.promises.writeFile(filePath, updatedJson, "utf-8");
+    console.log(`Field "shortLink" added successfully.`);
+  } catch (e) {
+    console.error("Error adding short link:", e);
+  }
+}
+
+main();
```
[the remark this answers is not in the document]
Comment (AndlerRL) [reply apps/web/scripts/generate-cms-dictionaries.ts:65]: @coderabbitai how can we refactor the code as you suggest by following your recommended changes?

**Comment by AndlerRL** [comment]
> Hey @leandrogavidia - I've reviewed your changes - here's some feedback:
> 
> **Overall Comments**:
> 
> * The '|| true' in the husky pre-commit script will cause it to always succeed even on failure, defeating the purpose of pre-commit hooks. Consider removing this.
> * Error handling in generate-cms-dictionaries.ts should fail the commit if critical operations fail, rather than just logging to console.
> 
> Here's what I looked at during the review
> Sourcery is free for open source - if you like our reviews please consider sharing them ✨
> Help me be more useful! Please click 👍 or 👎 on each comment and I'll use the feedback to improve your reviews.

Thanks, @sourcery-ai for the insights. @leandrogavidia , how sure is to generate the short link on pre-commit and these are being copied to the build folder properly? Maybe because of the dub.co error that you received is the cause of why some articles does not have a shortLink value?

@coderabbitai any additional insights? I noticed the vercel build went through, but also noticed that some static pages were built hence, I want to make sure if we are actually capturing the static json files from the dictionary 🤔 (we should though).

**Review by AndlerRL — CHANGES_REQUESTED** [review]
Changes looks good to me, the script seems to capture the missing JSON files for the static blog pages however, I notice some details that needs to be either fix or mark as resolved (not required). Please take a look on coderabbitai feedbacks and resolve them before pushing to develop. Also the additional note given before this review.

**Comment by AndlerRL** [comment]
> Tip
> 
> For best results, initiate chat on the files or code changes.
> 
> @AndlerRL: Based on your observations, I believe the issue might be related to how the `generate-cms-dictionaries.ts` script is being run in the pre-commit hook. Currently, the script is executed using `nohup` and `|| true`, which can cause it to run in the background and ignore any errors. This means the script may not finish before the commit proceeds, potentially leading to missing `shortLink` values in the static JSON files.
> 
> To ensure that the script runs synchronously and that any errors are properly handled, I recommend modifying the pre-commit script to remove `nohup` and `|| true`. Here's an updated version of the script:
> 
> ```shell
> #!/bin/bash
> echo "Running pre-commit scripts..."
> cd ./apps/web
> bun generate-cms-dictionaries.ts
> ```
> 
> By running the script synchronously, the commit process will wait for it to complete, and any errors will prevent the commit from proceeding. This should help ensure that the `shortLink` values are properly generated and that the static JSON files from the dictionary are accurately captured.
> 
> Let me know if you need any further assistance.

Thanks @coderabbitai . Would it then this await for each shortLink generation? What if it fails to generate a single link? shouldn't we add a guard and proceed with the rest?

@leandrogavidia this looks interesting. Let try this and see the results, I am wondering if we are quickly reaching the dub.co limit or it is a concurrence script execution issue 🤔

**Comment by leandrogavidia** [comment]
@AndlerRL Hi, please check the new update. I read the coderabbit's feedback and their conversation. 

Also, I added the validation so as not to add the link when a file already has it.

**Review by AndlerRL — APPROVED** [review]
LGTM! I made small modifications on the concurrency and the shortLink call. Merging to continue with the SSG scripts! :+1:

**Pull request merged**